docs/dev-tips.md

```diff
@@ -0,0 +1,97 @@
+* `sudo apt-get install libgtk-3-dev`
+* `sudo apt-get install libwebkit2gtk-4.0-dev`
+
+## Clone the project source code
+
+Clone the [GitHub repo here](https://github.com/mattermost/focalboard).
+
+## Build and run from the terminal
+
+Follow the steps in the [main readme file](https://github.com/mattermost/focalboard#building-the-server). In summary, to build and run the server:
+
+```
+make prebuild
+make
+ ./bin/focalboard-server
+```
+
+Then open a browser to `http://localhost:8000` to access it. The port is configured in `config.json`.
+
+Once the server is running, you can rebuild just the webapp with `make webapp` (in a separate terminal window), then reload the browser.
+
+## VSCode Setup
+
+Here's a recommended dev-test loop using VSCode:
+* Open a bash terminal window to the project folder
+* Run `make prebuild` to npm install
+    * Do this again when dependencies change in `webapp/package.json`
+* Run `cd webapp && npm run watchdev`
+    * This will auto-build the webapp on file changes
+* Open VSCode
+    * Install the [Go](https://marketplace.visualstudio.com/items?itemName=golang.Go) and [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) VSCode extensions if you haven't already
+* Hit F5 and select `Go: Launch Server`
+    * Or, press `Cmd+P` and type `debug <space>` and pick the option
+* Open a browser to `http://localhost:8000`
+    * The port is configured in `config.json`
+
+You can now edit the webapp code and refresh the browser to see your changes.
+
+## Debugging the web app
+
+You can use your favorite browser to debug the webapp code. With Chrome, open the dev tools with `Cmd+Alt+I` (`Ctrl+Alt+I` in Windows).
+* `npm run watchdev` builds the dev package, which includes source maps from js to typescript
+* In the Chrome devtools source tab, press `Cmd+P` to jump to a source file
+
+As a starting point, add a breakpoint to the `render()` function in `BoardPage.tsx`, then refresh the browser to walk through page rendering.
+
+## Debugging the server
+
+Debug the Go code in VSCode. This is set up automatically when you launch the server from there.
+
+To start, add a breakpoint to `handleGetBlocks()` in `server/api/api.go`, then refresh the browser to see how data is retrieved.
+
+## Localization / Internationalization / Translation
+
+We use `i18n` to localize the web app. Localized string generally use `intl.formatMessage`.
+
+When adding or modifying localized strings, run `npm run i18n-extract` in `webapp` to rebuild `webapp/i18n/en.json`.
+
+Translated strings are stored in other json files under `webapp/i18n`, e.g. `es.json` for Spanish.
+
+## Database
+
+By default, data is stored in a sqlite datebase `focalboard.db`. You can view and edit this directly using `sqlite3 focalboard.db` from bash.
+
+## Unit tests
+
+Before checking-in commits, run: `make ci`, which is simlar to the ci.yml workflow and includes:
+* Server unit tests: `make server-test`
```

Comment:
FYI - `make server-test` tests against all supported databases.  `make server-test-sqlite` runs just the Sqlite tests.